script.js

```diff
@@ -0,0 +1,235 @@
+      elemDiv.classList.add('slideshow__thumbnail-imgs');
+
+      elemImg = document.createElement('img');
+      elemImg.src = slide.imgSrc;
+      elemDiv.appendChild(elemImg);
+
+      elemDiv.addEventListener('click', (function (fakeSlide) {
+        return function () {
+          controler.setCurrentSlide(fakeSlide);
+          viewSlide.render();
+        }
+      })(slide));
+
+      this.elemThumbnails.appendChild(elemDiv);
+    }
+  }
+
+ };
+
+ var viewOptionBar = {
+    init: function () {
+      this.elemButtonInfo = document.querySelector('.slideshow__button--info');
+      this.elemButtonThumbnails = document.querySelector('.slideshow__button--thumbnails');
+      this.elemCounter = document.querySelector('.slideshow__counter');
+      this.elemButtonPrev = document.querySelector('.slideshow__button--prev');
+      this.elemButtonPlay = document.querySelector('#slideshow__playBtn');
+      this.elemButtonNext = document.querySelector('.slideshow__button--next');
+      var intervalSlides;
+
+      this.elemButtonNext.addEventListener('click', function () {
+        var slide = controler.getNextSlide();
+        controler.setCurrentSlide(slide);
+        viewSlide.render();
+        viewOptionBar.render();
+      });
+
+      this.elemButtonPrev.addEventListener('click', function () {
+        var slide = controler.getPrevSlide();
+        controler.setCurrentSlide(slide);
+        viewSlide.render();
+        viewOptionBar.render();
+      });
+
+      this.elemButtonInfo.addEventListener('click', function () {
+        if (viewThumbnail.elemThumbnails.style.display === 'block'){
+          viewThumbnail.elemThumbnails.style.display = 'none';
+        }
+        viewSlide.slideDescriptionElem.style.display = 
+          viewSlide.slideDescriptionElem.style.display === 'none'? 'block' : 'none';
+      });
+
+      this.elemButtonThumbnails.addEventListener('click', function () {
+        if (viewSlide.slideDescriptionElem.style.display === 'block'){
+          viewSlide.slideDescriptionElem.style.display = 'none';
+        }
+        viewThumbnail.elemThumbnails.style.display = 
+          viewThumbnail.elemThumbnails.style.display === 'none'? 'block' : 'none';
+      });
+
+      this.elemButtonPlay.addEventListener('click', function () {
+        viewOptionBar.elemButtonPlay.classList.toggle('slideshow__button--play');
+        viewOptionBar.elemButtonPlay.classList.toggle('slideshow__button--pause');
+        controler.setAutoPlay(!controler.getAutoPlay());
+        if(controler.getAutoPlay()){
+          viewSlide.autoRender();
+          intervalSlides = setInterval(viewSlide.autoRender, controler.getSLIDE_TIMEOUT());  
+        } else {
+          clearInterval(intervalSlides);
+        }
+        
+      });
+
+      this.render();
+    },
+
+    render: function () {
+      var counter = modal.currentSlide.idx + 1;
+      this.elemCounter.textContent = counter + "/" + modal.slides.length;
```

Comment:
this is really awesome to know https://twitter.com/_developit/status/997132016612270081